src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java

```diff
@@ -177,6 +178,11 @@ public void start() {
         .withName(name), DEFAULT_SELECTOR_THREAD_COUNT);
     workerScheduler = getWorkerScheduler(schedulersConfig.withName(name + ".requester.workers"));
 
+    redirectScheduler = schedulerService.customScheduler(schedulersConfig
+        .withDirectRunCpuLightWhenTargetBusy(true)
```

Comment:
wouldn't this lead back to the deadlock as the redirect would happen in the same thread?